packages/@lwc/shared/src/fields.ts

```diff
@@ -4,38 +4,35 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { create, isUndefined } from './language';
+import { isUndefined } from './language';
 import { hasNativeSymbolSupport } from './symbol';
 
-/*
- * The __type property on HiddenField doesn't actually exists at runtime. It is only used to store
- * the type of the associated field value.
- */
-export type HiddenField<T> = { __type: T };
-
-// This method abstracts the creation of symbols, so we can fallback to strings when native symbols
-// are not supported.
-export function createHiddenField<T = unknown>(key: string, namespace: string): HiddenField<T> {
-    return (hasNativeSymbolSupport ? Symbol(key) : `$$lwc-${namespace}-${key}$$`) as any;
-}
+export type HiddenField<T> = {
+    set: (o: any, value: T) => void;
+    get: (o: any) => T | undefined;
+};
 
-const hiddenFieldsMap: WeakMap<any, Record<any, any>> = new WeakMap();
+// Typically there are lots of objects we're assigning hidden fields on, but few unique fields.
+// So we create fewer objects and do less work to map from Field -> Object -> Value rather than
+// from Object -> Field -> Value
+const hiddenFieldMaps: Map<any, WeakMap<any, any>> = new Map();
 
-export function setHiddenField<T>(o: any, field: HiddenField<T>, value: T): void {
-    let valuesByField = hiddenFieldsMap.get(o);
+export function createHiddenField<T = unknown>(key: string, namespace: string): HiddenField<T> {
```

Comment:
why use `T = unknown` instead of just `T`?

```suggestion
export function createHiddenField<T>(key: string, namespace: string): HiddenField<T> {
```

Reply:
Because I'm bad at TypeScript. :joy: